pallets/nft/src/lib.rs

```diff
@@ -0,0 +1,412 @@
+        pub fn mint_nft(origin, ticker: Ticker, nft_metadata_attributes: Vec<NFTMetadataAttribute>, portfolio_kind: PortfolioKind) -> DispatchResult {
+            Self::base_mint_nft(origin, ticker, nft_metadata_attributes, portfolio_kind)
+        }
+
+        /// Burns the given NFT from the caller's portfolio.
+        ///
+        /// # Arguments
+        /// * `origin` - is a signer that has permissions to act as an agent of `ticker`.
+        /// * `ticker` - the ticker of the NFT collection.
+        /// * `nft_id` - the id of the NFT to be burned.
+        /// * `portfolio_kind` - the portfolio that contains the nft.
+        ///
+        /// ## Errors
+        /// - `CollectionNotFound` - if the collection associated to the given ticker has not been created.
+        /// - `NFTNotFound` - if the given NFT does not exist in the portfolio.
+        ///
+        /// # Permissions
+        /// * Asset
+        /// * Portfolio
+        #[weight = <T as Config>::WeightInfo::burn_nft(T::MaxNumberOfCollectionKeys::get() as u32)]
+        pub fn burn_nft(origin, ticker: Ticker, nft_id: NFTId, portfolio_kind: PortfolioKind) -> DispatchResult {
+            Self::base_burn_nft(origin, ticker, nft_id, portfolio_kind)
+        }
+    }
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> {
+        /// An overflow while calculating the balance.
+        BalanceOverflow,
+        /// An underflow while calculating the balance.
+        BalanceUnderflow,
+        /// The ticker is already associated to an NFT collection.
+        CollectionAlredyRegistered,
+        /// The NFT collection does not exist.
+        CollectionNotFound,
+        /// A duplicate metadata key has been passed as parameter.
+        DuplicateMetadataKey,
+        /// The associated asset is not of type NFT.
+        InvalidAssetType,
+        /// Either the number of keys or the key identifier does not match the keys defined for the collection.
+        InvalidMetadataAttribute,
+        /// Failed to transfer an NFT - NFT collection not found.
+        InvalidNFTTransferCollectionNotFound,
+        /// Failed to transfer an NFT - attempt to move to the same portfolio.
+        InvalidNFTTransferSamePortfolio,
+        /// Failed to transfer an NFT - NFT not found in portfolio.
+        InvalidNFTTransferNFTNotOwned,
+        /// Failed to transfer an NFT - balance would overflow.
+        InvalidNFTTransferBalanceOverflow,
+        /// Failed to transfer an NFT - not enough balance.
+        InvalidNFTTransferNoBalance,
+        /// Failed to transfer an NFT - compliance failed.
+        InvalidNFTTransferComplianceFailure,
+        /// The maximum number of metadata keys was exceeded.
+        MaxNumberOfKeysExceeded,
+        /// The NFT does not exist.
+        NFTNotFound,
+        /// At least one of the metadata keys has not been registered.
+        UnregisteredMetadataKey,
+    }
+}
+
+impl<T: Config> Module<T> {
+    fn base_create_nft_collection(
```

Comment:
If I understand the logic correctly, the following would be possible:
- call asset::create_asset(..., ACME, ..., AssetType::NFT, ...) [create a fungible asset with type NFT]
- call asset::issue(ACME, ...) [mint / distribute fungible asset with type NFT]
- call nft::create_nft_collection(..., ACME, ...) [create NFT with same ticker]
- call nft::mint(...) [mint NFTs]

Maybe I'm missing the logic that prevents this, but it would lead to Asset::balance_of etc. having a mix of fungible / non-fungible balances I think?